### DISCUSSION TITLE
`left_join` with a tuple of `Selectable` structs

### DISCUSSION THREAD

**Discussion by Jonathan-Landeed** [diesel-rs/diesel#3204] [Q&A]
I noticed I can make a tuple of selectable structs with `inner_join`, but I can't figure out how to do that with a left join. `nullable` wasn't working.

```
    #[derive(Queryable, Selectable)]
    struct A {
        foo: i32,
        b_id: Option<i32>,
    }

    #[derive(Queryable, Selectable)]
    struct B {
        bar: i32,
    }

    let conn = &mut connection();
    let data = table_a.inner_join(table_b).select((A::as_select(), B::as_select())).get_result(conn);  // works
    let data = table_a.left_join(table_b).select((A::as_select(), B::as_select())).get_result(conn);  // doesn't work
    let data = table_a.left_join(table_b).select((A::as_select(), B::as_select().nullable())).get_result(conn);  // doesn't work

```

I ended up creating a selectable struct with the left-joined struct embeded as an option, but is there a simpler way?

```
    #[derive(Queryable, Selectable)]
    struct A {
        foo: i32,
        #[diesel(embed)]
        b: Option<B>,
    }

    #[derive(Queryable, Selectable)]
    struct B {
        bar: i32,
    }

    let conn = &mut connection();
    let data = table_a.left_join(table_b).select(A::as_select()).get_result(conn);  // works
```

**Comment by weiznich** [comment]
I would call both errors expected behaviour. The first one cannot ever construct a valid select clause. The second one is not valid as `Selectable` is designed to lock in the return type, so modifying it later is not possible at all.

I would expect `.select((A::as_select(), Option::<B>::as_select()))` to work as well, but I haven't tested it yet.
